manifests/profile.pp

```diff
@@ -0,0 +1,22 @@
+define lvm::profile (
+  $volume,
+  $group,
+  $filename   = "${name}.profile",
+  $allocation = {},
+  $activation = {},
+) {
+
+  file { "/etc/lvm/profile/${filename}":
+    ensure  => file,
+    content => template('lvm/profile.erb'),
```

Comment:
LVM metadata profile file should be read only, so you can add a file mode option (`mode    => '0444'`).